Makefile

```diff
@@ -120,53 +201,146 @@ ${GOPATH}/out/yaml/crds: crds
+	kubectl create ns simple || /bin/true
+	# Global default may be strict or permissive - make it explicit for this ns
+	kubectl -n simple apply -f test/k8s/mtls_permissive.yaml
+	kubectl -n simple apply -f test/k8s/sidecar-local.yaml
+	(set -o pipefail; cd ${GOPATH}/src/istio.io/istio; \
+	  go test -v -timeout 25m ./tests/e2e/tests/simple -args \
+	     --auth_enable=false \
+         --egress=false --ingress=false \
+         --rbac_enable=false --cluster_wide \
+         --skip_setup \
+         --use_local_cluster=true \
+         --istio_namespace=${ISTIO_NS} \
+         --namespace=simple   \
+         --istioctl=${ISTIOCTL_BIN} \
+           2>&1 | tee ${GOPATH}/out/logs/$@.log)
+
+# Simple test, strict mode
+run-simple-strict: ${TMPDIR}
+	kubectl create ns simple-strict || /bin/true
+	kubectl -n simple-strict apply -f test/k8s/mtls_strict.yaml
+	kubectl -n simple apply -f test/k8s/sidecar-local.yaml
+	(cd ${GOPATH}/src/istio.io/istio; make e2e_simple_run ${TEST_FLAGS} \
+		E2E_ARGS="${E2E_ARGS} --namespace=simple-strict")
+
+run-bookinfo-demo:
+	kubectl create ns bookinfo-demo || /bin/true
+	kubectl -n simple apply -f test/k8s/mtls_permissive.yaml
+	kubectl -n simple apply -f test/k8s/sidecar-local.yaml
+	(cd ${GOPATH}/src/istio.io/istio; make e2e_bookinfo_run ${TEST_FLAGS} \
+		E2E_ARGS="${E2E_ARGS} --namespace=bookinfo-demo")
+
+run-mixer:
+	kubectl create ns mixertest || /bin/true
+	kubectl -n mixertest apply -f test/k8s/mtls_permissive.yaml
+	kubectl -n mixertest apply -f test/k8s/sidecar-local.yaml
+	(cd ${GOPATH}/src/istio.io/istio; make e2e_mixer_run ${TEST_FLAGS} \
+		E2E_ARGS="${E2E_ARGS} --namespace=mixertest")
+
+INT_FLAGS ?= -istio.test.hub ${HUB} -istio.test.tag ${TAG} -istio.test.pullpolicy IfNotPresent \
+ -p 1 -istio.test.env kube --istio.test.kube.config ${KUBECONFIG} --istio.test.ci --istio.test.nocleanup \
+ --istio.test.kube.deploy=false -istio.test.kube.systemNamespace ${ISTIO_NS} -istio.test.kube.minikube
+
+# Integration tests create and delete istio-system
+# Need to be fixed to use new installer
+# Requires an environment with telemetry installed
+run-test.integration.kube:
+	export TMPDIR=${GOPATH}/out/tmp
+	mkdir -p ${GOPATH}/out/tmp
+	#(cd ${GOPATH}/src/istio.io/istio; \
+	#	$(GO) test -v ${T} ./tests/integration/... ${INT_FLAGS})
+	kubectl -n default apply -f test/k8s/mtls_permissive.yaml
+	kubectl -n default apply -f test/k8s/sidecar-local.yaml
+	# Test uses autoinject
+	#kubectl label namespace default istio-env=${ISTIO_NS} --overwrite
+	(cd ${GOPATH}/src/istio.io/istio; TAG=${TAG} make test.integration.kube \
+		CI=1 T="-v" K8S_TEST_FLAGS="--istio.test.kube.minikube --istio.test.kube.systemNamespace ${ISTIO_NS} \
+		 --istio.test.nocleanup --istio.test.kube.deploy=false ")
+
+run-test.integration.kube.presubmit:
+	export TMPDIR=${GOPATH}/out/tmp
+	mkdir -p ${GOPATH}/out/tmp ${GOPATH}/out/linux_amd64/release/ ${GOPATH}/out/logs/
+	set -o pipefail; \
+	cd ${GOPATH}/src/istio.io/istio; \
```

Comment:
just for my education, how does istio.io/istio know where to locate the installer?

Reply:
It doesn't - the flags on the tests are used to skip the installation of istio ( skip setup / deploy ), so the tests run against whatever the installer has setup.